### PR TITLE
ESM and CJS dual package support

### PR DESCRIPTION
**Pull request by h3poteto**

### CODE REVIEW

File: megalodon/tsconfig.mjs.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "module": "ES2020",
```
Comment (ThatOneCalculator) [inline megalodon/tsconfig.mjs.json:4]: Is there any reason that ES2020 is being used instead of ES2023 or ESNext?

**Review by ThatOneCalculator — COMMENTED** [review]
Maybe `browser` should be changed to `webpack`, since both Webpack and Vite target the browser

File: megalodon/tsconfig.mjs.json (hunk 1 above)
Comment (h3poteto) [reply megalodon/tsconfig.mjs.json:4]: There is no reason. Is ES2023 better?

Comment (ThatOneCalculator) [reply megalodon/tsconfig.mjs.json:4]: Yes, it's generally best to keep the ES module version either as the current year or `ESNext`

**Pull request merged**